src/test/java/org/mockito/internal/matchers/AssertionMatcherTest.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (C) 2015 Mockito contributors.
+ *
+ * Licensed under the Apache License, Version 2.0.
+ */
+package org.mockito.internal.matchers;
+
+import net.bytebuddy.ClassFileVersion;
+import org.assertj.core.api.ThrowableAssert;
+import org.assertj.core.api.WithAssertions;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.internal.matchers.domain.ShipSearchCriteria;
+import org.mockito.internal.matchers.domain.TacticalStation;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+import static net.bytebuddy.ClassFileVersion.JAVA_V9;
+import static org.junit.Assume.assumeTrue;
+import static org.mockito.AdditionalMatchers.assertArg;
+import static org.mockito.Mockito.verify;
+
+public class AssertionMatcherTest implements WithAssertions {
+
+    @Rule
+    public MockitoRule mockitoRule = MockitoJUnit.rule();
+
+    @Mock
+    private TacticalStation ts;
+
+    private ShipSearchCriteria searchCriteria = new ShipSearchCriteria(1000, 4);
```

Comment:
In my project I use those two classes also for some other use cases and in fact it can be too complex for just `AssertionMatcherTest`.